pkg/compact/compact.go

```diff
@@ -667,6 +667,55 @@ func (ds *DownsampleProgressCalculator) ProgressCalculate(ctx context.Context, g
 	return nil
 }
 
+// RetentionProgressMetrics contains Prometheus metrics related to retention progress.
+type RetentionProgressMetrics struct {
+	NumberOfBlocksDeleted *prometheus.GaugeVec
```

Comment:
`NumberOfBlocksDeleted` uses past tense but the progress is future tense. Let's rename it to `NumberOfBlocksToDelete` or `NumberOfBlocksRetention`?